include/deal.II/base/symmetric_tensor.h

```diff
@@ -1016,19 +1022,9 @@ namespace internal
 
 
 
-template <int rank_, int dim, typename Number>
-inline SymmetricTensor<rank_, dim, Number>::SymmetricTensor()
-{
-  // Some auto-differentiable numbers need explicit
-  // zero initialization.
-  for (unsigned int i = 0; i < base_tensor_type::dimension; ++i)
-    data[i] = internal::NumberType<Number>::value(0.0);
```

Comment:
The comments indicate that there are `Number` types that won't work correctly with a default constructor, see https://github.com/dealii/dealii/commit/97d3e0bc01d.

Reply:
Looking at the tests, I found that these tests already check the default constructor and the copy constructor for `SymmetricTensor`.
 * `adolc/symmetric_tensor_functions_01_1`
 * `adolc/symmetric_tensor_functions_01_2`
 * `sacado/symmetric_tensor_functions_01_1`
 * `sacado/symmetric_tensor_functions_01_2`
 * `sacado/symmetric_tensor_functions_01_3`
 * `sacado/symmetric_tensor_functions_01_4`

These tests check every aspect of symmetric tensor functions. I am going to modify my constexpr tests (tensors/constexpr_symmetric_tensor.cc) to cover similar functions.

Since the private member `data` is a `Tensor`, it already has a well-defined default constructor for any number type.

Reply:
Yes, I didn't mean for you to add tests. Sorry for the confusion!
AFAICT, the testers don't run the `Sacado` tests and I was implying that we should do so manually.

Reply:
16 tests using symmetric tensor with Sacado types pass on my machine. (GCC 7.4, Trilinos 12.10.1)

Reply:
OK, thanks.